subsys/bluetooth/host/hci_core.c

```diff
@@ -6697,7 +6697,7 @@ int bt_set_name(const char *name)
 		return 0;
 	}
 
-	strncpy(bt_dev.name, name, sizeof(bt_dev.name));
+	strncpy(bt_dev.name, name, len);
```

Comment:
This looks like the wrong fix to me. If `len` is less than the previous name you'll end up getting a corrupted name, since there will not be a null terminator after the new name (the string will terminate wherever the old name did). Instead, the right fix is to pass `sizeof(bt_dev.name) - 1` as the last parameter, or alternatively use `strlcpy()` but I don't think we have that function available.

Reply:
`strlcpy` is not available in the minimal lib C, but we could of course add it.

Using `sizeof(bt_dev.name) - 1` we could be copying more than `name` though, maybe just put the null-terminator in ourselves?
```
strncpy(bt_dev.name, name, len);
bt_dev.name[len] = '\0';
```

If using `sizeof(bt_dev.name) - 1` maybe CONFIG_BT_DEVICE_NAME_MAX is better?
And `if (len > CONFIG_BT_DEVICE_NAME_MAX)` in the max length check above?

Reply:
> Using sizeof(bt_dev.name) - 1 we could be copying more than name though

How so? `name` is assumed to be null-terminated and`strncpy()` would stop at the null character.